pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,251 @@
+	if !accoutIsmemberOfOrg && !reValidatedcache {
+		orgAccountList, err := o.refreshCache(ctx, orgClient)
+		if err != nil {
+			return false, err
+		}
+		_, accoutIsmemberOfOrg = orgAccountList[accoundIDOfNode]
+	}
+
+	return accoutIsmemberOfOrg, nil
+}
+
+// If cache miss happens, refresh list with new cache and check if element exist
+func (o *orgValidator) refreshCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (map[string]any, error) {
+	remTries := o.getRetries()
+
+	orgAccountList := make(map[string]any)
+	if remTries <= 0 {
+		return orgAccountList, nil
+	}
+
+	orgAccountList, err := o.reloadAccountList(ctx, orgClient, true)
+	if err != nil {
+		return nil, err
+	}
+
+	o.decrRetries()
+
+	return orgAccountList, nil
+}
+
+// checkIfOrgAccountListIsStale checks if the cached org account list is stale.
+func (o *orgValidator) checkIfOrgAccountListIsStale(_ context.Context) bool {
+	o.mutex.RLock()
+	defer o.mutex.RUnlock()
+
+	// Map is empty that means this is first time plugin is being initialised
+	if len(o.orgListAccountMap) == 0 {
+		return true
+	}
+
+	// Get the timestamp from config
+	existingTimestamp := o.orgListAccountMapCreationTime
+
+	return checkIfTTLIsExpired(existingTimestamp, o.orgAccountListCacheTTL)
+}
+
+// reloadAccountList gets the list of accounts belonging to organization and catch them
+func (o *orgValidator) reloadAccountList(ctx context.Context, orgClient organizations.ListAccountsAPIClient, catchBurst bool) (map[string]any, error) {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	// Make sure: we are not doing cache burst and account map is not updated recently from different go routine.
+	if !catchBurst && len(o.orgListAccountMap) != 0 && !checkIfTTLIsExpired(o.orgListAccountMapCreationTime, o.orgAccountListCacheTTL) {
+		return o.orgListAccountMap, nil
+	}
+
+	// Avoid if other thread has already updated the map
+	if catchBurst && o.retries == 0 {
+		return o.orgListAccountMap, nil
+	}
+
+	// Get the list of accounts
+	listAccountsOp, err := orgClient.ListAccounts(ctx, &organizations.ListAccountsInput{})
+	if err != nil {
+		return nil, fmt.Errorf("issue while getting list of accounts: %w", err)
+	}
+
+	// Build new org accounts list
+	orgAccountsMap := make(map[string]any)
+
+	// Update the org account list cache with ACTIVE accounts & handle pagination
+	for {
+		for _, acc := range listAccountsOp.Accounts {
+			if acc.Status == types.AccountStatusActive {
+				accID := *acc.Id
+				orgAccountsMap[accID] = true
+			}
+		}
+
+		if listAccountsOp.NextToken == nil {
+			break
+		}
+
+		listAccountsOp, err = orgClient.ListAccounts(ctx, &organizations.ListAccountsInput{
+			NextToken: listAccountsOp.NextToken,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("issue while getting list of accounts in pagination: %w", err)
+		}
+	}
+
+	// Update timestamp, if it was not invoked as part of cache miss.
+	if !catchBurst {
+		t := time.Now().UTC()
+		o.orgListAccountMapCreationTime = t
```

Comment:
what do you think, about calling it
```
o.notAfter = o.clock.Now().Add(o. orgAccountListCacheTTL)
```

Soo we can check if not expired doing something like:

```
expired := s.clock.Now().After(s.notAfter)
```